nixos/tests/systemd-cryptenroll.nix

```diff
@@ -0,0 +1,43 @@
+let
+  etpm = "/tmp/emulated_tpm";
+in
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "systemd-cryptenroll";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ ymatsiuk ];
+  };
+
+  machine = { pkgs, lib, ... }: {
+    environment.systemPackages = [ pkgs.cryptsetup ];
+    virtualisation = {
+      emptyDiskImages = [ 512 ];
+      memorySize = 1024;
+      qemu.options = [
+        "-chardev socket,id=chrtpm,path=${etpm}/swtpm-sock"
+        "-tpmdev emulator,id=tpm0,chardev=chrtpm"
+        "-device tpm-tis,tpmdev=tpm0"
+      ];
+    };
+  };
+
+  testScript = ''
+    import subprocess
+
+    subprocess.run(["mkdir", "-p", "${etpm}"])
+    # start TPM2 emulation as a separate process
+    subprocess.Popen(["${pkgs.swtpm}/bin/swtpm", "socket", "--tpmstate", "dir=${etpm}", "--ctrl", "type=unixio,path=${etpm}/swtpm-sock", "--log", "level=0", "--tpm2"])
+
+    start_all()
+    machine.succeed("test -e /dev/tpm0")
+    machine.succeed("test -e /dev/tpmrm0")
+    machine.succeed("systemd-cryptenroll --tpm2-device=list")
+    machine.succeed("echo -n lukspass | cryptsetup luksFormat -q /dev/vdb -")
+    machine.succeed("PASSWORD=lukspass systemd-cryptenroll --tpm2-device=auto --tpm2-pcrs=7 /dev/vdb")
+    machine.fail("PASSWORD=wrongpass systemd-cryptenroll --tpm2-device=auto --tpm2-pcrs=7 /dev/vdb")
+    machine.succeed("${pkgs.systemd}/lib/systemd/systemd-cryptsetup attach luks /dev/vdb - tpm2-device=auto")
+    machine.succeed("test -e /dev/mapper/luks")
+    machine.succeed("${pkgs.systemd}/lib/systemd/systemd-cryptsetup detach luks")
+    machine.succeed("systemd-cryptenroll --wipe-slot=tpm2 /dev/vdb")
```

Comment:
Can we add this (or another partition) to `/etc/crypttab`, and see if unlocking via that works? (similar to nixos/tests/systemd.nix, L146)